### ISSUE TITLE
[DocDB] SELECT statement throwing error while backup creation is under process, with cross region geo-partitioning

### ISSUE THREAD

**Issue by agsh-yb** [yugabyte/yugabyte-db#17217]
Jira Link: [DB-6468](https://yugabyte.atlassian.net/browse/DB-6468)
### Description

- This issue is seen in `2.18`, also in `master`
- Executing `SELECT COUNT(*) FROM tab1;` While Backup is being created, We get Error:
```
ERROR: Wrong number of responses: 0, while 1 expected: 0 vs 1
```
- Create an universe with cross region geo-partitioning 

![Screenshot 2023-05-08 at 11 56 06 AM](https://user-images.githubusercontent.com/121818428/236754642-f777e744-f8e2-4055-ade7-105a45db3de8.png)

Steps:
```
1. CREATE DATABASE db;
2. Create tablespace
CREATE TABLESPACE us_west_2_tablespace WITH (
  replica_placement='{"num_replicas": 1, "placement_blocks":
  [
  {"cloud":"aws","region":"us-west-2","zone":"us-west-2a","min_num_replicas":1}]}'
);

CREATE TABLESPACE us_east_1_tablespace WITH (
  replica_placement='{"num_replicas": 1, "placement_blocks":
  [
  {"cloud":"aws","region":"us-east-1","zone":"us-east-1a","min_num_replicas":1}]}'
);

CREATE TABLESPACE ap_south_1_tablespace WITH (
  replica_placement='{"num_replicas": 1, "placement_blocks":
  [
  {"cloud":"aws","region":"ap-south-1","zone":"ap-south-1a","min_num_replicas":1}]}'
);

4. Create Extension

CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
CREATE TYPE complex AS (re float8, im float8);
CREATE TYPE e_details AS ENUM ('Email', 'Sms', 'Phone');
CREATE DOMAIN postal_code AS TEXT CHECK(VALUE ~ '^\d{5}$'OR VALUE ~ '^\d{5}-\d{4}$');

5. Create table

CREATE TABLE tab1 (id TEXT, uuid_col uuid DEFAULT uuid_generate_v4(), name text COLLATE "en_US.utf8", c complex, info json, contact JSONB, arr smallint[], cash money, i inet, m macaddr, i2 serial, i3 bigserial, val smallint, details e_details, age int, collated_data text collate "POSIX",date DATE, n NUMERIC (3, 2),r real, c1 CHAR(1), created_at timestamptz, uuid0 uuid DEFAULT uuid_nil(), uuid1 uuid DEFAULT uuid_generate_v1(), p1 POINT,t1 TIME,ts1 TIMESTAMP,i4 INTERVAL,p2 path, p3 polygon,b box, c2 circle, l line, l1 lseg, a2 text[][], zip postal_code, geo_partition VARCHAR, PRIMARY KEY (id HASH, name, geo_partition)) PARTITION BY LIST (geo_partition);

6. Create Partition Table

CREATE TABLE tab1_us_east
    PARTITION OF tab1
      (id, uuid_col, name, c, info, contact, arr, cash, i, m, i2, i3, val, details, age, collated_data,date, n,r, c1, created_at, uuid0, uuid1, p1,t1,ts1,i4,p2, p3,b, c2, l, l1, a2, zip, geo_partition)
    FOR VALUES IN ('USEAST') TABLESPACE us_east_1_tablespace;

CREATE TABLE tab1_india
    PARTITION OF tab1
      (id, uuid_col, name, c, info, contact, arr, cash, i, m, i2, i3, val, details, age, collated_data,date, n,r, c1, created_at, uuid0, uuid1, p1,t1,ts1,i4,p2, p3,b, c2, l, l1, a2, zip, geo_partition)
    FOR VALUES IN ('India') TABLESPACE ap_south_1_tablespace;

CREATE TABLE tab1_us_west
    PARTITION OF tab1
      (id, uuid_col, name, c, info, contact, arr, cash, i, m, i2, i3, val, details, age, collated_data,date, n,r, c1, created_at, uuid0, uuid1, p1,t1,ts1,i4,p2, p3,b, c2, l, l1, a2, zip, geo_partition)
    FOR VALUES IN ('USWEST') TABLESPACE us_west_2_tablespace;

7. Load Data

DO
$do$
BEGIN 
   FOR i IN 1..100001 LOOP
      INSERT INTO tab1(id, name, c, info, contact, arr, cash, i, m, i2, i3,val,details,age,collated_data, date,n,r,c1,created_at,p1,t1,ts1,i4,p2,p3,c2,l,l1,a2,zip,geo_partition) VALUES (i,'Hermione',(1,2), '{"customer": "John Doe", "items": {"product": "Laptop","qty": 6}}','{"phones":[ {"type": "mobile", "phone": "001001"}, {"type": "fix", "phone": "002002"}]}', '{1, 2, 3}', '$99.99' ,'1.1.1.1', '00:00:00:00:00:00',i,i+1,666,'Sms',15, md5(random()::text), current_timestamp,5.36,2147483647,'G',now(), point(3, 4),LOCALTIME(0), current_timestamp(0),  '00:25:00', '(1,3), (4,12)','(1,3),(4,12), (2,4)','10, 4, 10','{1,2,3}','(0, 4), (2,8)','{{"b1", "c"}, {"m1", "l"}}',89000, 'USWEST');
   END LOOP;
END
$do$;

8. Create Backup
9. While Backup is being created,

 SELECT count(*) FROM tab1;
```
After backup is created SELECT query is working fine.

### Warning: Please confirm that this issue does not contain any sensitive information

- [X] I confirm this issue does not contain any sensitive information.

[DB-6468]: https://yugabyte.atlassian.net/browse/DB-6468?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by kripasreenivasan** [comment]
FYI @lingamsandeep

**Comment by rthallamko3** [comment]
@tverona1 , @m-iancu , It seems like this failure occurs even when backup is not running.

**Comment by Arjun-yb** [comment]
@rthallamko3 @lingamsandeep @m-iancu @tverona1 
I found similar error message with COPY cmd
Version: `2.18.0.0-b61`
```
copy bkup_restore_ysql_table_range(id,name,c,info,contact,arr,cash,i,m,i2,i3,val,details,age,collated_data,date,n,r,c1,created_at,p1,t1,ts1,i4,p2,p3,b,c2,l,l1,a2,zip) from '/tmp/ysql_example1.ysql' with delimiter '#';

ERROR:  Wrong number of responses: 0, while 3072 expected: 0 vs 3072
```
Here, I am copying 100K rows, and it failed with above error after copying 20K rows to table.
Then I tried same copy command to another table in the DB and found that it is failing immediately with same error

```
copy bkup_restore_ysql_table_hash(id,name,c,info,contact,arr,cash,i,m,i2,i3,val,details,age,collated_data,date,n,r,c1,created_at,p1,t1,ts1,i4,p2,p3,b,c2,l,l1,a2,zip) from '/tmp/ysql_example1.ysql' with delimiter
 '#';
ERROR:  Wrong number of responses: 0, while 3072 expected: 0 vs 3072
```

**Comment by fizaaluthra** [comment]
fixed by commit eb6fae3f3d21f18c2884e18696a4e8b1b093b9f4 (2.18 backport cdfc59351ab736968b9f09d84fc9bd655270c311)
